etc/templates/assertions_entry_point_class_template.txt

```diff
@@ -0,0 +1,16 @@
+package ${package};
+
+/**
+ * Entry point for assertions of different data types. Each method in this class is a static factory for the
+ * type-specific assertion objects.
+ */
+@javax.annotation.Generated(value="assertj-assertions-generator")
+public class Assertions extends org.assertj.core.api.Assertions {
```

Comment:
I am not convinced we want to accompain junit/hamcrest with assertj. Do you feel strongly that we should (I did not gave assertJ a try, yet)?

Reply:
Well, maybe this is a matter of taste. I use AssertJ in all my projects and teach students that this is the best assertion library. I think with AssertJ you get the best fluent code for tests at no additional costs. Especially you do not need to remember the assertions as you get every valid assertion on code completion. AssertJ produces clickable failures that open the diff dialog in IntelliJ and I think it cooperates with Page Objects quite well. 

I think the tests are more readable as well: 
AssertJ:
```
        build.open();
        AnalysisSummary checkstyle = new AnalysisSummary(build, CHECKSTYLE_ID);
        assertThat(checkstyle).isDisplayed();
        assertThat(checkstyle).hasTitleText("CheckStyle: 3 warnings");
        assertThat(checkstyle).hasNewSize(3);
        assertThat(checkstyle).hasFixedSize(1);
        assertThat(checkstyle).hasReferenceBuild(1);
        assertThat(checkstyle).hasInfoType(InfoType.ERROR);

        AnalysisResult checkstyleDetails = checkstyle.openOverallResult();
        assertThat(checkstyleDetails).hasActiveTab(Tab.CATEGORIES);
        assertThat(checkstyleDetails).hasTotal(3);
        assertThat(checkstyleDetails).hasOnlyAvailableTabs(Tab.CATEGORIES, Tab.TYPES, Tab.ISSUES);

```
Harmcrest:
```
        assertThat(action.getNumberOfWarnings(), is(count));
        assertThat(action.getNumberOfNewWarnings(), is(count));
        assertThat(action.getNumberOfFixedWarnings(), is(0));
        assertThat(action.getNumberOfWarningsWithHighPriority(), is(0));
        assertThat(action.getNumberOfWarningsWithNormalPriority(), is(count));
        assertThat(action.getNumberOfWarningsWithLowPriority(), is(0));

        assertThat(driver, hasContent(title + ": 12"));

        Map<String, Integer> warningsPerAxis = new HashMap<>();
        warningsPerAxis.put("user_axis=one", 4);
        warningsPerAxis.put("user_axis=two", 6);
        warningsPerAxis.put("user_axis=three", 2);
        for (MatrixConfiguration axis : job.getConfigurations()) {
            Build axisBuild = axis.getLastBuild();
            assertThat(axisBuild, hasAction(title));
            assertThat(driver, hasContent(title + ": " + warningsPerAxis.get(axis.name)));
        }
```

But if this is too much change I can remove it.